tests/test_basics.py

```diff
@@ -1024,3 +1024,19 @@ class Nested:
     f = Nested()
     assert {"v": {"v": 100}} == serde.to_dict(f)
     assert f == serde.from_dict(Nested, {})
+
+
+def test_nested_dataclass_without_serde() -> None:
+    @dataclasses.dataclass
+    class Foo:
+        v: int
+
+    @serde.serde
+    @dataclasses.dataclass
+    class Wrapper:
+        foo: Foo
```

Comment:
Do you also want to serialize/deserialize `Wrapper` without `@serde`? 🤔 
Maybe it's possible if we checks and generate code in [from_obj](https://github.com/yukinarit/pyserde/blob/2582e244642dc1ba1df68f7ff04018b018db735c/serde/de.py#L352) and [to_obj](https://github.com/yukinarit/pyserde/blob/2582e244642dc1ba1df68f7ff04018b018db735c/serde/se.py#L323)? What do you think?

Reply:
I was also thinking about this. It already sort-of works for serialization of dataclasses, although it just uses `dataclasses.asdict`, so it only works for basic stuff.

Not having `@serde` on the "root" type is not that important to me specifically, but I think that it should be supported for consistency. It seems slightly orthogonal though, do you want me to implement it in this PR or in a follow-up one?

Reply:
I am ok with the follow-up PR!